File: src/controller/BoardBasedCellListener.java
public class BoardBasedCellListener implements CellListener {

    private final Board board;

    public BoardBasedCellListener(Board board) {
        this.board = board;
    }

    public void cellClicked(Cell cell) {
        board.makeTurn(cell.getX(), cell.getY());
    }
}

File: src/controller/BoardBasedCellListener.kt
class BoardBasedCellListener(private val board: Board) : CellListener {

    override fun cellClicked(cell1: Cell, cell2: Cell) {
        board.makeShip(cell1.x, cell1.y, cell2.x, cell2.y)
    }

    override fun killship(cell: Cell) {
        board.killship(cell.x, cell.y)
    }
}
